expertsystem/data.py

```diff
@@ -238,6 +243,53 @@ def __repr__(self) -> str:
     def add(self, particle: Particle) -> None:
         self.__particles[particle.name] = particle
 
+    def find(self, search_term: Union[int, str]) -> Particle:
+        """Search for a particle by either name (`str`) or PID (`int`)."""
+        if isinstance(search_term, str):
+            particle_name = search_term
+            return self.__particles[particle_name]
+        if isinstance(search_term, int):
+            pid = search_term
+            search_results = [
+                particle for particle in self.values() if particle.pid == pid
+            ]
+            if len(search_results) == 0:
+                raise LookupError(f"Could not find particle with PID {pid}")
+            if len(search_results) > 1:
```

Comment:
I believe a 
```python3
if search_results:
.... 
else:
...
```
statement would be more elegant here

Reply:
I don't think so, the method finalises because it raises an exception. For sure, `pylint` would complain about an `else` here.

Reply:
Ah nvm I misread that code